docs/data/charts/export/ExportChartAsImage.tsx

```diff
@@ -73,7 +73,7 @@ function ExportParamsSelector({
 }
 
 export default function ExportChartAsImage() {
-  const apiRef = React.useRef<ChartProApi>(undefined);
+  const apiRef = React.useRef<ChartProApi<'line'>>(undefined);
```

Comment:
Unfortunately, I don't think we can keep the current API for `ChartProApi`. Either we create a new type, or we need to accept this breaking change as a bug fix. 

It is a bug because invalid types are accepted. For example, the heatmap's `apiRef` doesn't contain a `setZoomData` property, but the type says it does. 

It's unlikely a lot of people are using it as it was [released in v8.1.0](https://github.com/mui/mui-x/commit/03408e35a8ff6240b65136e1120b20dbf7ca757a), but it's technically a breaking change if we don't flag it as a bug. 

Another option would be to create another type to avoid breaking users and deprecate this one. What are your thoughts?

Reply:
We could just go with the generic one but still allow users to narrow the types with the `<'seriesType'>` notation.

Reply:
That doesn't work either. I suppose it's the same issue as trying to pass a `RefObject<HTMLElement>` to a `HTMLDivElement`.

Reply:
I don't get it, if the issue is that you are passing a wide type but it expects a narrow type, just make the expectation wide, so it can accept a narrow one.

Reply:
I think I managed to make it work. 

I thought it was another case of this issue with `HTMLDivElement` and `HTMLElement`:

![image](https://github.com/user-attachments/assets/e4eda08c-d40c-4b98-83f3-307c17cd5c49)

I've kept the `ChartProApi<'line'>` here because we should encourage users to use it, but it's usable without it as well:

![image](https://github.com/user-attachments/assets/83ff1481-e773-4efe-8656-e30fe6ae0473)